bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -1348,7 +1350,7 @@ public ImageData getImageData() {
 public ImageData getImageData (int zoom) {
 	if (isDisposed()) SWT.error(SWT.ERROR_GRAPHIC_DISPOSED);
 
-	if (zoom == currentDeviceZoom) {
+	if (zoom == getZoom()) {
```

Comment:
Similar to https://github.com/eclipse-platform/eclipse.platform.swt/pull/1229#discussion_r1601379823, please extract `getZoom()` to a variable called `oldZoom` to improve readability. It makes sense in this case since the method receives a (new/targeted) `zoom` as parameter and the method `getZoom()` retrieves the current/soon-to-be-old zoom.

Reply:
in this case we would call it currentZoom since here we are not setting the new zoom but requesting imageData at a particular zoom. I'd propose to call it currentZoom. The logic is like if the zoom is equal to currentZoom of the Image then return the imageData of the image otherwise do some caling and other stuff.